schemas.js

```diff
@@ -80,7 +80,60 @@ yup.addMethod(yup.string, "schemaFileExists", function (srcDir) {
   });
 });
 
+// validateServiceName - Custom validation method to validate service name
+yup.addMethod(yup.string, "validateServiceName", function () {
+  return this.test({
+    name: "validate-service-name",
+    test: (value, testCtx) => {
+      const alphanumericRegex = "[a-z0-9_-]+";
+      const choreoSvcRefNameRegex = new RegExp(
+        `^choreo:\/\/\/${alphanumericRegex}\/${alphanumericRegex}\/${alphanumericRegex}\/${alphanumericRegex}\/v\\d+(\\.\\d+)?\/(PUBLIC|PROJECT|ORGANIZATION)$`
+      );
+      const thirdPartySvcRefNameRegex = new RegExp(
+        "^thirdparty:[a-zA-Z0-9._/-]+$"
+      );
+      const dbSvcRefNameRegex = new RegExp("^database:[a-z0-9_/-]+$");
+
+      if (value.startsWith("choreo:///")) {
+        return (
+          choreoSvcRefNameRegex.test(value) ||
+          new yup.ValidationError(
+            `${testCtx.path} must follow the format ` +
+              `choreo:///<org-handle>/<project-handle>/<component-handle>/<endpoint-identifier>/<major-version>/<network-visibility>`
+          )
+        );
+      }
+      if (value.startsWith("thirdparty:")) {
+        return (
+          thirdPartySvcRefNameRegex.test(value) ||
+          new yup.ValidationError(
+            `${testCtx.path} has an invalid service identifier, ` +
+              `only alphanumeric characters, periods (.), underscores (_), hyphens (-), and slashes (/) are allowed after thirdparty:`
+          )
+        );
+      }
+      if (value.startsWith("database:")) {
+        return (
+          dbSvcRefNameRegex.test(value) ||
+          new yup.ValidationError(
+            `${testCtx.path} has an invalid service identifier, ` +
+              `only alphanumeric characters, underscores (_), hyphens (-), and slashes (/) are allowed after database:`
+          )
+        );
+      }
+      return (
+        thirdPartySvcRefNameRegex.test(value) ||
+        dbSvcRefNameRegex.test(value) ||
+        new yup.ValidationError(
+          `${testCtx.path} has an invalid service type. It can only contain choreo, thirdparty, or database types.`
+        )
+      );
```

Comment:
Do we need the Yup validations here again? Since we do early returns if any of the predefined checks pass, this return will get executed only if its an invalid service type.

If we're adding this to ensure all scenarios are covered, Shall we also include the missing `choreo` service name validation?

Reply:
Ah yes i missed this, the last return we can just return the validation error, no need to do this `thirdPartySvcRefNameRegex.test(value) ||
        dbSvcRefNameRegex.test(value) ||`
 Updated